crates/sqlbuiltins/src/functions/table/bson.rs

```diff
@@ -65,16 +64,9 @@ impl TableFunc for BsonScan {
 
         let (source_url, storage_options) = table_location_and_opts(ctx, args, &mut opts)?;
 
-        let url = match source_url {
-            DatasourceUrl::File(path) => DatasourceUrl::File(resolve_path(&path)?),
-            DatasourceUrl::Url(_) => source_url,
-        };
-
-        let store_access = GenericStoreAccess::new_from_location_and_opts(
-            url.to_string().as_str(),
-            storage_options,
-        )?;
+        let store_access = storage_options_into_store_access(&source_url, &storage_options)
+            .map_err(ExtensionError::access)?;
```

Comment:
> sort of feels like we should just combine the table_location_and_opts with storage_options_into_store_access into one function as neither is particularly useful without the other.

I don't think so. `table_location_and_opts` is used only for function implementations.

> Also, can we just implement the appropriate From<> for the storage_options_into_store_access error type?

Sure! Will do.

Reply:
> Also, can we just implement the appropriate From<> for the storage_options_into_store_access error type?

So, we can't really do that because this is a trait method returning `ExtensionError` (which lies in `datafusion_ext` crate).

Reply:
alas!
 
makes sense.